egg_updater.py

```diff
@@ -0,0 +1,62 @@
+import os
+import sys
+from io import StringIO
+
+from setuptools.dist import Distribution
+
+
+def IS_PRODUCTION():
+    return any(
+        list(
+            map(
+                os.path.isfile,
+                list(
+                    map(
+                        lambda syspath: os.path.join(syspath, "cve-bin-tool.egg-link"),
+                        sys.path,
+                    )
+                ),
+            )
+        )
+    )
+
+
+def update_egg():
+    with StringIO() as f:
+        sys.stdout = f
+        dist = Distribution(
```

Comment:
We should probably do a `try: os.chdir` to `os.path.join(os.path.dirname(__file__), "..")` and then a `finally: os.chdir(wherever_os.getcwd()_reported_before_first_chdir)`

Reply:
Yes this will work.

Reply:
I have tried it, it's not working

Reply:
What's not working?

Reply:
In the above example: `os.path.join(os.path.dirname(__file__), "..")` should be `os.path.dirname(__file__)`

Reply:
I have put egg_updater inside cve_bin_tool and save cwd by `cwd = os.getcwd()` and then changed diectory to `os.chdir(os.path.dirname(__file__))` and after execution of script, I have reverted back to the previous `cwd` but it is giving same error as we get without applying `chdir` hack.   

```console
(venv) C:\Users\Root\Documents\cve-bin-tool>python -m cve_bin_tool.cli build
cve_bin_tool - WARNING -
                          **********************************************
                          Warning: this utility was developed for Linux.
                          You may need to install additional utilities
                          to use it on other operating systems.
                          **********************************************

cve_bin_tool.CVEDB - INFO - Using cached CVE data (<24h old). Use -u now to update immediately.
cve_bin_tool.Scanner - INFO - Updating egg_info
C:\Users\Root\Documents\cve-bin-tool C:\Users\Root\Documents\cve-bin-tool\cve_bin_tool
Traceback (most recent call last):
  File "C:\Users\Root\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\Root\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Root\Documents\cve-bin-tool\cve_bin_tool\cli.py", line 587, in <module>
    sys.exit(main())
  File "C:\Users\Root\Documents\cve-bin-tool\cve_bin_tool\cli.py", line 471, in main
    scanner = Scanner(cvedb)
  File "C:\Users\Root\Documents\cve-bin-tool\cve_bin_tool\cli.py", line 72, in __init__
    egg_updater.update_egg()
  File "C:\Users\Root\Documents\cve-bin-tool\cve_bin_tool\egg_updater.py", line 35, in update_egg
    "checkers",
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\Root\\Documents\\cve-bin-tool\\cve_bin_tool\\cve_bin_tool\\c
heckers'

```

Reply:
It's solved.